xtra-libp2p-offer/src/current/maker.rs

```diff
@@ -111,6 +119,22 @@ impl NewOffers {
 #[derive(Clone, Copy)]
 pub struct GetLatestOffers;
 
+#[derive(Clone, Default)]
+struct Offers(HashMap<(ContractSymbol, Position), model::Offer>);
```

Comment:
What am I missing that you chose a hashmap here if you always use `to_vec` before consuming it?

Reply:
The point of the map is to ensure that we only have one offer per `contract_symbol`-`position_maker` pair. That is, when we get new ones we always replace the old ones. This is done on line 129.

If we stored it as a list we would send unnecessary and conflicting data to the newly connected takers. They would be capable of dealing with it the way we implemented the taker side. But it's not much work for the maker to do this.

Also if we just kept appending to a list the size of the message for newly connected takers would keep growing towards infinity over time 😆

Reply:
Fair enough :) 
I guess you could achieve the same thing with a vec: finding the offer in it and replacing it if found or adding it to the vec.

Reply:
Absolutely. I'm just assuming that inserting into a map is faster than finding and replacing into a vector.